apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-logo/edit.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { IconButton, ServerSideRender, Toolbar } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import { BlockControls } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+function SiteLogoEdit( { className } ) {
+	const siteIdentityUrl = addQueryArgs( 'customize.php', {
+		'autofocus[section]': 'title_tagline',
+		return: window.location.href,
+	} );
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<Toolbar>
+					<IconButton icon="edit" label={ __( 'Edit Site Logo' ) } href={ siteIdentityUrl } />
```

Comment:
I think this will need some styles now to match other toolbar buttons: https://github.com/Automattic/wp-calypso/pull/33622/commits/467b6bbe5b1c6a4553b678bf9a109c6ec4260e13